support/modules/rootfs-image

```diff
@@ -30,15 +33,28 @@ else
     passive_num=$MENDER_ROOTFS_PART_A_NUMBER
 fi
 passive_num_hex=$(printf '%x' $passive_num)
+upgrade_available="$(${PRINTENV} upgrade_available)"
+upgrade_available="${upgrade_available#upgrade_available=}"
 
 ensure_correct_root_mounted() {
-    if [ "$(stat -L -c %02t%02T $active)" != "$(stat -L -c %04D /)" ]; then
-        echo "Mounted root does not match boot loader environment!" 1>&2
-        exit 1
+    if [ "$(stat -L -c %02t%02T $active)" = "$(stat -L -c %04D /)" ]; then
+        return 0
     fi
+
+    # If an upgrade is in progress then it can be on either partition, except in the Download
+    # and ArtifactInstall states (not rebooted yet).
+    if [ "$upgrade_available" = 1 ] && \
```

Comment:
function?

Reply:
I think it's better without a function actually, since it relies on a variable. If we make a function, then either we have to pass it in, or use the global variable which then makes it less obvious where the variable is used.